### DISCUSSION TITLE
Build MIT/EPL/... versions with github and gradle

### DISCUSSION THREAD

**Discussion by arnaudroques** [plantuml/plantuml#1336] [General]
PlantUML is provided under several licenses. Right now, only the GPL version is hosted on github.
[Other versions](https://plantuml.com/download)  (LGPL, MIT, EPL, ASL...) are hosted on sourceforge.

Recently, we create a tiny preprocessor [sjpp](https://github.com/plantuml/sjpp) for [the Javascript CheerpJ version]( https://github.com/plantuml/plantuml-core)

Now we can use this preprocessor to generate the other licenses version. We have added some `__MIT__` comment in the main source. So it's now possible to generate sources for the MIT version (basically, we remove all code that cannot be release under MIT license).

We have set up specific CI [ci-mit.yml](https://github.com/plantuml/plantuml/blob/master/.github/workflows/ci-mit.yml)  and gradle [build-mit.gradle.kts](https://github.com/plantuml/plantuml/blob/master/build-mit.gradle.kts) file to show [it working](https://github.com/plantuml/plantuml/actions/runs/4441397058/jobs/7796415495).

The preprocessor can be run as [simple jar](https://github.com/plantuml/plantuml/blob/master/.github/workflows/ci-mit.yml#L36) or plain ant task, which means that is could be simply run from gradle.

However, we are not very confortable with neither github action nor gradle.
And we have probably not done it in the right way.

So we call for help :-)
@soloturn, @gliptak and @The-Lum , you have help us successfully. If you have some time some days, it would be nice that if you could clean our proposal up to the [publication of the MIT version](https://github.com/plantuml/plantuml/releases).
We would then generalize this to all others licenses.

What do you think about it?
Many thanks for your support!

**Comment by travkin79** [comment]
Hello @arnaudroques,

Has there been any progress concerning building the differently licensed plantuml lib files? Is there someone, who might help us with that? That would help us building new PlantUML Eclipse plug-in releases (see [discussion](https://github.com/hallvard/plantuml/discussions/166)).

Best regards,
Dietrich

**Comment by travkin79** [comment]
Hi @arnaudroques,

It seems, your scripts are working already. I've activated building the other differently licensed PlantUML versions, too. Would you please review and merge the PR #1471?
I think, improving the scripts could still be done later, right?

Thank you. Best regards.
Dietrich

**Comment by travkin79** [comment]
**Update**
Activated releasing plantuml versions for additional licenses: PRs #1471, #1473
